src/fairchem/core/models/base.py

```diff
@@ -57,111 +54,45 @@ def generate_graph(
-            # Not all models will have this attribute
-            enforce_max_neighbors_strictly = self.enforce_max_neighbors_strictly
-        else:
-            # Default to old behavior
-            enforce_max_neighbors_strictly = True
-
-        if not otf_graph:
-            try:
-                edge_index = data.edge_index
-
-                if use_pbc:
-                    cell_offsets = data.cell_offsets
-                    neighbors = data.neighbors
-
-            except AttributeError:
-                logging.warning(
-                    "Turning otf_graph=True as required attributes not present in data object"
-                )
-                otf_graph = True
-
-        if use_pbc:
-            if otf_graph:
-                if use_pbc_single:
-                    (
-                        edge_index_per_system,
-                        cell_offsets_per_system,
-                        neighbors_per_system,
-                    ) = list(
-                        zip(
-                            *[
-                                radius_graph_pbc(
-                                    data[idx],
-                                    cutoff,
-                                    max_neighbors,
-                                    enforce_max_neighbors_strictly,
-                                )
-                                for idx in range(len(data))
-                            ]
-                        )
-                    )
-
-                    # atom indexs in the edge_index need to be offset
-                    atom_index_offset = data.natoms.cumsum(dim=0).roll(1)
-                    atom_index_offset[0] = 0
-                    edge_index = torch.hstack(
-                        [
-                            edge_index_per_system[idx] + atom_index_offset[idx]
-                            for idx in range(len(data))
-                        ]
-                    )
-                    cell_offsets = torch.vstack(cell_offsets_per_system)
-                    neighbors = torch.hstack(neighbors_per_system)
-                else:
-                    ## TODO this is the original call, but blows up with memory
-                    ## using two different samples
-                    ## sid='mp-675045-mp-675045-0-7' (MPTRAJ)
-                    ## sid='75396' (OC22)
-                    edge_index, cell_offsets, neighbors = radius_graph_pbc(
-                        data,
-                        cutoff,
-                        max_neighbors,
-                        enforce_max_neighbors_strictly,
-                    )
-
-            out = get_pbc_distances(
-                data.pos,
-                edge_index,
-                data.cell,
-                cell_offsets,
-                neighbors,
-                return_offsets=True,
-                return_distance_vec=True,
+        def choose_attr_from_input_or_self(attr, input_args, default=None):
```

Comment:
nit; Can we rename input_args to input_arg ? its only the single arg not a set of them right?

Reply:
Thanks and fixed!